sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/ResolveDefaultColumnsUtil.scala

```diff
@@ -241,4 +243,33 @@ object ResolveDefaultColumns {
       }
     }
   }
+
+  /**
+   * Returns a new Analyzer for processing default column values using built-in functions only.
```

Comment:
IIUC the analyzer is initialized only once, since it is a `val` of an `object`

Reply:
Thanks for pointing this out, I updated the comment.